### ISSUE TITLE
annotate_neo_termini of data-raw/pkd_mouse_model_search/protein.fas lead to Error in `left_join()`

### ISSUE THREAD

**Issue by a-matte** [MiguelCos/TermineR#3]
Hi,

I have been trying to use your pipeline which seems amazing. However, running this function :

annotated_df_quant <- annotate_neo_termini(
  peptides_df = df_from_fragpipe, 
  fasta_location = here::here(".../data-raw/pkd_mouse_model_search/protein.fas"),
  sense = "C", 
  specificity = "K|R",
  organism = "mouse") %>% 
  mutate(cleav_len = nchar(cleavage_sequence)) %>% 
  relocate(cleav_len, .after = cleavage_sequence)

I receive the following error message:

Error in `left_join()`:
! `...` must be empty.
✖ Problematic argument:
• relationship = "many-to-many"
Run `rlang::last_trace()` to see where the error occurred.

> rlang::last_trace()
<error/rlib_error_dots_nonempty>
Error in `left_join()`:
! `...` must be empty.
✖ Problematic argument:
• relationship = "many-to-many"
---
Backtrace:
    ▆
 1. ├─... %>% relocate(cleav_len, .after = cleavage_sequence)
 2. ├─dplyr::relocate(., cleav_len, .after = cleavage_sequence)
 3. ├─dplyr::mutate(., cleav_len = nchar(cleavage_sequence))
 4. └─TermineR::annotate_neo_termini(...)
 5.   ├─dplyr::left_join(...)
 6.   └─dplyr:::left_join.data.frame(protein_nter, df_mol_proc_feat, by = "protein", relationship = "many-to-many")
 

protein.fas has been directly downloaded from your repository.

Thank you for your work,
Arthur

**Comment by MiguelCos** [comment]
Hello Arthur, 

I am glad you are using the tool and thanks for reporting this.

Can you share more information on what you did so I can try to reproduce the error?

Are you using the sample data or your own input data? If you are using your own input, would you be wiling to share it with me so I can do some tests?

Thanks.

Miguel

**Comment by a-matte** [comment]
Hi, 

I have been trying to follow your “Extended example” tutorial (https://github.com/MiguelCos/TermineR/blob/main/terminomics_analysis_workflow.md). I downloaded the three files in your repository data-raw/pkd_mouse_model_search: annotation.txt, protein.fas and psm.tsv.

The first steps worked well until the step Annotation of peptide specificities. There this function blocked as described above:

# Execute the annotation function
annotated_peptides <- annotate_neo_termini(
  peptides_df = df_from_fragpipe,
  fasta_location = fasta_location,
  sense = "C",
  specificity = "K|R", # sence C and specificity K|R for trypsin digestion
  organism = "mouse"
)


I’m workink on  R version 4.2.1 (2022-06-23 ucrt) on Windows and here are the versions of the loaded packages:
RColorBrewer version: 1.1-3
pheatmap version: 1.0.12
SummarizedExperiment version: 1.26.1
GenomicRanges version: 1.48.0
GenomeInfoDb version: 1.32.4
MatrixGenerics version: 1.8.1
matrixStats version: 1.3.0
ggrepel version: 0.9.5
ggsignif version: 0.6.4
ggpubr version: 0.6.0
seqinr version: 4.2-36
janitor version: 2.2.0
here version: 1.0.1
org.Mm.eg.db version: 3.15.0
AnnotationDbi version: 1.58.0
IRanges version: 2.30.1
S4Vectors version: 0.34.0
Biobase version: 2.56.0
BiocGenerics version: 0.42.0
clusterProfiler version: 4.4.4
limma version: 3.52.4
lubridate version: 1.9.3
forcats version: 1.0.0
stringr version: 1.5.1
dplyr version: 1.1.0
purrr version: 1.0.1
readr version: 2.1.5
tidyr version: 1.3.0
tibble version: 3.1.8
ggplot2 version: 3.5.1
tidyverse version: 2.0.0
TermineR version: 1.0.0

All the best, 
Arthur

**Comment by a-matte** [comment]
Hi,

I cloned your repo, and while it didn't work initially, I was able to get it running by updating my version of dplyr. I successfully ran your terminomics_analysis_workflow.qmd script without any issues after that. It seems that dplyr version 1.1.0 doesn't work with your pipeline, but version 1.1.4 does.

Thank you very much for your help, I will happily try to implement your approach in my current project.

Best,
Arthur
